controllers/etcd_custodian_controller.go

```diff
@@ -130,7 +131,9 @@ func (ec *EtcdCustodian) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, err
 	}
 
-	if err := ec.updatePodDisruptionBudget(ctx, logger, etcd, refMgr); err != nil {
+	pdbValues := componentpdb.GenerateValues(etcd)
+	pdbDeployer := componentpdb.New(ec.Client, etcd.Namespace, &pdbValues, ec.restConfig)
+	if err := pdbDeployer.Deploy(ctx); err != nil {
```

Comment:
why is PDB being now created from 2 controllers?

Reply:
`Deploy` does two things, creates a PDB and updates it. 
Here, it is intended to keep the PDB up to date
Custodian controller won't start it's job until the sts is up. Etcd controller creates the PDB before creating the sts

Reply:
Custodian controller should only update the status and not the object, right? Here there is a potential to actually create/update the PDB causing 2 actors calling `Deploy`. Is there any other resource for which similar thing is done in the custodian controller?

Reply:
No, there is no other resource for which a similar thing is done in custodian controller

However, I've updated the PR to address this concern. The PDB component now has a method to only sync the `minAvailable` field that's used by the custodian controller